docs/plugins/development/creating-text-analysis-plugins.asciidoc

```diff
@@ -0,0 +1,129 @@
+
+Elastic provides a Grade plugin, `elasticsearch.stable-esplugin`, that makes it
+easier to develop and package plugins. The steps below assume you use this 
+plugin. However, you don't need Gradle to create plugins.
+
+The {es} Github repository contains
+{es-repo}tree/main/plugins/examples/stable-analysis[an example analysis plugin].
+The example `build.gradle` build script provides a good starting point for 
+developing your own plugin.
+
+[discrete]
+===== Prerequisites
+
+Plugins are written in Java, so you need to install a Java Development Kit
+(JDK). Install Gradle if you want to use Gradle.
+
+[discrete]
+===== Step by step
+
+. Create a directory for your project. 
+. Copy the example `build.gradle` build script to your project directory.  Note
+that this build script uses the `elasticsearch.stable-esplugin` gradle plugin to
+build your plugin.
+. Edit the `build.gradle` build script:
+** Add a definition for the `elasticsearchVersion` and matching `luceneVersion` 
+variables to the top of the file. You can find these versions in the 
+`build-tools-internal/version.properties` file in the {es-repo}[Elasticsearch 
+Github repository].
+** Edit the `name` and `description` in the `esplugin` section of the build 
+script.
+** Add module information.
+** Ensure you have declared the following compile-time dependencies. These 
+dependencies are compile-time only because {es} will provide these libraries at 
+runtime.
+*** `org.elasticsearch.plugin:elasticsearch-plugin-api`
+*** `org.elasticsearch.plugin:elasticsearch-plugin-analysis-api`
+*** `org.apache.lucene:lucene-analysis-common`
+** For unit testing, ensure these dependencies have also been added to the 
+`build.gradle` script as `testImplementation` dependencies.
+. Implement an interface from the analysis plugin API, annotating it with 
+`NamedComponent`. Refer to <<example-text-analysis-plugin>> for an example.
+. You should now be able to assemble a plugin ZIP file by running:
++
+[source,sh]
+----
+gradle bundlePlugin
+----
+The resulting plugin ZIP file will be written to the  `build/distributions` 
+directory.
+
+[discrete]
+===== YAML REST tests
+
+The Gradle `elasticsearch.yaml-rest-test` plugin enables testing of your 
+plugin using the {es-repo}blob/main/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc[{es} yamlRestTest framework]. 
+These tests use a YAML-formatted domain language to issue REST requests against
+an internal {es} cluster that has your plugin installed, and to check the 
+results of those requests. The structure of a YAML REST test directory is as 
+follows:
+** A test suite class, defined under `src/yamlRestTest/java`. This class should extend
+`ESClientYamlSuiteTestCase`.
+** If the plugin defines new REST endpoints, the specifications for those 
+endpoints should be defined in `src/yamlRestTest/resources/rest-api-spec/api/`.
+** The YAML tests themselves should be defined under `src/yamlRestTest/resources/test/`.
```

Comment:
re ** 
was this used to create a bullet list? it did not render as a list in a preview

Reply:
Good catch! I've fixed this.